grails-app/services/io/xh/toolbox/App/NewsService.groovy

```diff
@@ -0,0 +1,88 @@
+package io.xh.toolbox.App
+
+import io.xh.hoist.json.JSON
+import io.xh.toolbox.BaseService
+import io.xh.toolbox.NewsItem
+import org.grails.web.json.JSONArray
+import org.grails.web.json.JSONObject
+
+import static io.xh.hoist.util.DateTimeUtils.MINUTES
+
+class NewsService extends BaseService {
+
+    public List<NewsItem> _newsItems
+
+    static clearCachesConfigs = ['newsSources', 'newsApiKey']
+    def configService
+
+    void init() {
+        log.info('news service initialized')
+        createTimer(
+                runFn: this.&loadAllNews,
+                interval: 10 * MINUTES
+        )
+        super.init()
+    }
+
+    List<NewsItem> getNewsItems() {
+        return _newsItems ? _newsItems.sort{-it.published.time} : Collections.emptyList()
+    }
+
+    void clearCaches() {
+        _newsItems = []
+        super.clearCaches()
+
+        loadAllNews()
+    }
+
+
+    //------------------------
+    // Implementation
+    //------------------------
+    private void loadAllNews() {
+        def sources = configService.getJSONObject('newsSources', new JSONObject())
```

Comment:
We can remove the default value fallback here.  We want to use those very sparingly, and in the special case of Toolbox we know we will have it installed in Bootstrap.  In general, though, we prefer these things to throw and be obvious about it if they don't have the right configs setup.

(I realize this default was likely there in the old PIQ code.)